commit 73c9ce37b15a963c6e609906d232b0a6ea9c741f
Author: Steve Grubb <sgrubb@redhat.com>
Date:   Wed Nov 1 17:22:47 2023 -0400

    declare file local function static

diff --git a/lib/libaudit.c b/lib/libaudit.c
index 72b25a9..cfbad1d 100644
--- a/lib/libaudit.c
+++ b/lib/libaudit.c
@@ -997,7 +997,7 @@ uint32_t audit_get_session(void)
 		return ses;
 }
 
-int audit_rule_syscall_data(struct audit_rule_data *rule, int scall)
+static int audit_rule_syscall_data(struct audit_rule_data *rule, int scall)
 {
 	int word = AUDIT_WORD(scall);
 	int bit  = AUDIT_BIT(scall);
