diff -urp audit-3.0.3.orig/src/Makefile.in audit-3.0.3/src/Makefile.in
--- audit-3.0.3.orig/src/Makefile.in	2021-06-28 18:04:56.450429090 -0400
+++ audit-3.0.3/src/Makefile.in	2021-06-28 18:05:37.325424715 -0400
@@ -487,7 +487,7 @@ auditd_SOURCES = auditd.c auditd-event.c
 	$(am__append_1)
 auditd_CFLAGS = -fPIE -DPIE -g -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pthread -Wno-pointer-sign
 auditd_LDFLAGS = -pie -Wl,-z,relro -Wl,-z,now
-auditd_LDADD = @LIBWRAP_LIBS@ ${top_builddir}/src/libev/libev.la ${top_builddir}/audisp/libdisp.la ${top_builddir}/lib/libaudit.la ${top_builddir}/auparse/libauparse.la -lpthread -lm -lrt $(gss_libs) ${top_builddir}/common/libaucommon.la
+auditd_LDADD = @LIBWRAP_LIBS@ ${top_builddir}/src/libev/libev.la ${top_builddir}/audisp/libdisp.la ${top_builddir}/lib/libaudit.la ${top_builddir}/auparse/libauparse.la -lpthread $(gss_libs) ${top_builddir}/common/libaucommon.la
 auditctl_SOURCES = auditctl.c auditctl-llist.c delete_all.c auditctl-listing.c
 auditctl_CFLAGS = -fPIE -DPIE -g -D_GNU_SOURCE
 auditctl_LDFLAGS = -pie -Wl,-z,relro -Wl,-z,now
