diff -urp audit-1.7.9.orig/src/ausearch-report.c audit-1.7.9/src/ausearch-report.c
--- audit-1.7.9.orig/src/ausearch-report.c	2008-10-24 13:49:39.000000000 -0400
+++ audit-1.7.9/src/ausearch-report.c	2008-10-24 15:24:51.000000000 -0400
@@ -63,7 +63,7 @@ static void output_interpreted_node(cons
 static void interpret(char *name, char *val, int comma, int rtype);
 
 /* The machine based on elf type */
-static int machine = 0;
+static int machine = -1;
 
 /* The first syscall argument */
 static unsigned long long a0;
@@ -424,7 +424,7 @@ static void print_syscall(const char *va
 	const char *sys;
 	int ival;
 
-	if (machine <= 0) 
+	if (machine < 0) 
 		machine = audit_detect_machine();
 	if (machine < 0) {
 		printf("%s ", val);
