diff -urp audit-2.7.2/src/ausearch-report.c audit-2.7.3/src/ausearch-report.c
--- audit-2.7.2/src/ausearch-report.c	2017-02-13 09:22:49.618094389 -0500
+++ audit-2.7.3/src/ausearch-report.c	2017-02-13 10:55:50.000000000 -0500
@@ -484,7 +484,7 @@ static void csv_event(auparse_state_t *a
 		fprintf(stderr, "error normalizing %s\n", type);
 
 		// Just dump an empty frame
-		printf(",,,,,,,,%s%s\n", extra_labels ? ",," : "",
+		printf(",,,,,,,,,%s%s\n", extra_labels ? ",," : "",
 			extra_keys ? "," : "");
 		return;
 	}
