commit 59c886a671c53741399fe9dea710c2bf1ae3d8f4
Author: Steve Grubb <sgrubb@redhat.com>
Date:   Wed Nov 1 15:40:32 2023 -0400

    use correct arg in audit_add_perm_syscalls

diff --git a/lib/libaudit.c b/lib/libaudit.c
index 563cc2f..0a52285 100644
--- a/lib/libaudit.c
+++ b/lib/libaudit.c
@@ -1522,7 +1522,7 @@ static int audit_add_perm_syscalls(int perm, struct audit_rule_data *rule)
 		_audit_syscalladded = 1;
 		break;
 	case -1: // Should never happen
-		audit_msg(LOG_ERR, "Syscall name unknown: %s", syscall);
+		audit_msg(LOG_ERR, "Syscall name unknown: %s", syscalls);
 		break;
 	default: // Error reported - do nothing here
 		break;
