Index: /trunk/lib/libaudit.h
===================================================================
--- /trunk/lib/libaudit.h (revision 171)
+++ /trunk/lib/libaudit.h (revision 234)
@@ -227,4 +227,12 @@
 
 #define AUDIT_FIRST_CRYPTO_MSG		2400
+#define AUDIT_CRYPTO_TEST_USER		2400 /* Crypto test results */
+#define AUDIT_CRYPTO_PARAM_CHANGE_USER	2401 /* Crypto attribute change */
+#define AUDIT_CRYPTO_LOGIN		2402 /* Logged in as crypto officer */
+#define AUDIT_CRYPTO_LOGOUT		2403 /* Logged out from crypto */
+#define AUDIT_CRYPTO_KEY_USER		2404 /* Create,delete,negotiate */
+#define AUDIT_CRYPTO_FAILURE_USER	2405 /* Fail decrypt,encrypt,randomiz */
+#define AUDIT_CRYPTO_REPLAY_USER	2406 /* Crypto replay detected */
+
 #define AUDIT_LAST_CRYPTO_MSG		2499
 
Index: /trunk/lib/msg_typetab.h
===================================================================
--- /trunk/lib/msg_typetab.h (revision 171)
+++ /trunk/lib/msg_typetab.h (revision 234)
@@ -175,3 +175,10 @@
 _S(AUDIT_FS_RELABEL,                 "FS_RELABEL"                    )
 _S(AUDIT_USER_MAC_POLICY_LOAD,       "USER_MAC_POLICY_LOAD"          )
+_S(AUDIT_CRYPTO_TEST_USER,           "CRYPTO_TEST_USER"              )
+_S(AUDIT_CRYPTO_PARAM_CHANGE_USER,   "CRYPTO_PARAM_CHANGE_USER"      )
+_S(AUDIT_CRYPTO_LOGIN,               "CRYPTO_LOGIN"                  )
+_S(AUDIT_CRYPTO_LOGOUT,              "CRYPTO_LOGOUT"                 )
+_S(AUDIT_CRYPTO_KEY_USER,            "CRYPTO_KEY_USER"               )
+_S(AUDIT_CRYPTO_FAILURE_USER,        "CRYPTO_FAILURE_USER"           )
+_S(AUDIT_CRYPTO_REPLAY_USER,         "CRYPTO_REPLAY_USER"            )
 
