diff -urN audit/system-config-audit/ChangeLog audit-1.5.5/system-config-audit/ChangeLog
--- audit/system-config-audit/ChangeLog	2007-06-27 15:07:59.000000000 +0200
+++ audit-1.5.5/system-config-audit/ChangeLog	2007-07-17 01:20:02.000000000 +0200
@@ -1,3 +1,8 @@
+2007-07-17  Miloslav Trmač  <mitr@redhat.com>
+
+	* system-config-audit.desktop.in (Categories): Add System, to move the
+	menu entry to the Administration submenu.
+
 2007-06-27  Miloslav Trmač  <mitr@redhat.com>
 
 	* configure.ac: Release 0.4.2.
diff -urN audit/system-config-audit/system-config-audit.desktop.in audit-1.5.5/system-config-audit/system-config-audit.desktop.in
--- audit/system-config-audit/system-config-audit.desktop.in	2007-05-15 00:41:31.000000000 +0200
+++ audit-1.5.5/system-config-audit/system-config-audit.desktop.in	2007-07-17 01:20:02.000000000 +0200
@@ -6,4 +6,4 @@
 Terminal=false
 Type=Application
 Icon=logviewer
-Categories=Settings;
\ Chybí znak konce řádku na konci souboru
+Categories=Settings;System;
