--- audit/src/auditd.c.reconfig	2007-07-01 04:12:43.000000000 +0200
+++ audit/src/auditd.c	2007-07-01 04:25:41.000000000 +0200
@@ -522,10 +522,8 @@ int main(int argc, char *argv[])
 							AUDIT_DAEMON_CONFIG, 
 				"auditd error getting hup info - no change,"
 				" sending auid=? pid=? subj=? res=failed");
-						    } else {
-							free(rep);
-							rep = NULL;
 						    }
+						    rep = NULL;
 						    hup_info_requested = 0;
 						} else if(usr1_info_requested){
 							char usr1[MAX_AUDIT_MESSAGE_LENGTH];
