diff -urp audit-2.8.4.orig/src/auditd.c audit-2.8.4/src/auditd.c
--- audit-2.8.4.orig/src/auditd.c	2018-05-21 13:38:08.000000000 -0400
+++ audit-2.8.4/src/auditd.c	2018-06-27 08:12:44.027716750 -0400
@@ -388,7 +388,7 @@ static int become_daemon(void)
 	if (do_fork) {
 		if (pipe(init_pipe) || 
 				fcntl(init_pipe[0], F_SETFD, FD_CLOEXEC) ||
-				fcntl(init_pipe[0], F_SETFD, FD_CLOEXEC))
+				fcntl(init_pipe[1], F_SETFD, FD_CLOEXEC))
 			return -1;
 		pid = fork();
 	} else
@@ -432,7 +432,9 @@ static int become_daemon(void)
 			break;
 		default:
 			/* Wait for the child to say its done */
-			rc = read(init_pipe[0], &status, sizeof(status));
+			do {
+				rc = read(init_pipe[0], &status,sizeof(status));
+			} while (rc < 0 && errno == EINTR);
 			if (rc < 0)
 				return -1;
 
diff -urp audit-2.8.4.orig/src/auditd-dispatch.c audit-2.8.4/src/auditd-dispatch.c
--- audit-2.8.4.orig/src/auditd-dispatch.c	2018-05-21 13:38:08.000000000 -0400
+++ audit-2.8.4/src/auditd-dispatch.c	2018-06-27 08:13:27.590712076 -0400
@@ -47,7 +47,6 @@ int dispatcher_pid(void)
 
 void dispatcher_reaped(void)
 {
-	audit_msg(LOG_INFO, "dispatcher %d reaped", pid);
 	pid = 0;
 	shutdown_dispatcher();
 }
